test/AWS.Deploy.CLI.IntegrationTests/ServerModeTests.cs

```diff
@@ -89,6 +93,61 @@ public async Task GetRecommendations()
             }
         }
 
+        [Fact]
+        public async Task GetRecommendationsWithEncryptedCredentials()
+        {
+            var projectPath = Path.GetFullPath(Path.Combine("testapps", "WebAppNoDockerFile", "WebAppNoDockerFile.csproj"));
+            var portNumber = 4000;
+
+            var aes = Aes.Create();
+            aes.GenerateKey();
+            aes.GenerateIV();
+
+            using var httpClient = ServerModeHttpClientFactory.ConstructHttpClient(ResolveCredentials, aes);
+
+            InMemoryInteractiveService interactiveService = new InMemoryInteractiveService();
+            var keyInfo = new EncryptionKeyInfo
+            {
+                Version = EncryptionKeyInfo.VERSION_1_0,
+                Key = Convert.ToBase64String(aes.Key),
+                IV = Convert.ToBase64String(aes.IV)
+            };
```

Comment:
Key and IV is correctly initialized here on creating the instance of `Aes`. However in case of CLI, we are reading from stdin and they can be null.

POCO also have Key and IV nullable
```chsarp
    public class EncryptionKeyInfo
    {
        public const string VERSION_1_0 = "1.0";

        /// <summary>
        /// The version of the key info. If the property is set to a value that server mode has not implemented to support then
        /// a fatal exception is thrown.
        /// </summary>
        public string? Version { get; set; }

        /// <summary>
        /// Encryption key base 64 encoded 
        /// </summary>
        public string? Key { get; set; }

        /// <summary>
        /// Encryption IV base 64 encoded
        /// </summary>
        public string? IV { get; set; }

        public static EncryptionKeyInfo ParseStdInKeyInfo(string input)
        {
            var json = Encoding.UTF8.GetString(Convert.FromBase64String(input));
            var keyInfo = Newtonsoft.Json.JsonConvert.DeserializeObject<EncryptionKeyInfo>(json);

            if(string.IsNullOrEmpty(keyInfo.Key))
            {
                throw newwInvalidEncryptionKeyInfoException("Encryption key info is missing \"Key\" property.");
            }

            return keyInfo;
        }
    }
```
we should have some mechanism of saying these (IV and Key) are minimum properties to be passed in order it to work.